src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIQuarrier.java

```diff
@@ -0,0 +1,725 @@
+            {
+                setBlockFromInventory(pos, getMainFillBlock());
+            }
+        }
+
+        final BlockState blockState = world.getBlockState(blockToMine);
+        if (!IColonyManager.getInstance().getCompatibilityManager().isOre(blockState))
+        {
+            blockToMine = getSurroundingOreOrDefault(blockToMine);
+        }
+
+        if (world.getBlockState(blockToMine).getBlock() instanceof AirBlock)
+        {
+            blockToMine = null;
+            return BUILDING_STEP;
+        }
+
+        if (!mineBlock(blockToMine, getCurrentWorkingPosition()))
+        {
+            worker.swing(Hand.MAIN_HAND);
+            return getState();
+        }
+
+        blockToMine = getSurroundingOreOrDefault(blockToMine);
+        if (IColonyManager.getInstance().getCompatibilityManager().isOre(world.getBlockState(blockToMine)))
+        {
+            return getState();
+        }
+
+        worker.decreaseSaturationForContinuousAction();
+        blockToMine  = null;
+        return BUILDING_STEP;
+    }
+
+    private BlockPos getSurroundingOreOrDefault(final BlockPos pos)
+    {
+        for (Direction direction : Direction.values())
+        {
+            final BlockPos offset = pos.relative(direction);
+            if (IColonyManager.getInstance().getCompatibilityManager().isOre(world.getBlockState(offset)))
+            {
+                return offset;
+            }
+        }
+        return pos;
+    }
+
+    /**
+     * Get the main fill block. Based on the settings.
+     *
+     * @return the main fill block.
+     */
+    private Block getMainFillBlock()
+    {
+        return getOwnBuilding().getSetting(FILL_BLOCK).getValue().getBlock();
+    }
+
+    @Override
+    public ItemStack getTotalAmount(final ItemStack stack)
+    {
+        if (ItemStackUtils.isEmpty(stack))
+        {
+            return null;
+        }
+
+        final ItemStack copy = stack.copy();
+        copy.setCount(Math.max(super.getTotalAmount(stack).getCount(), copy.getMaxStackSize() / 2));
+        return copy;
+    }
+
+    @Override
+    public IAIState afterStructureLoading()
+    {
+        return BUILDING_STEP;
+    }
+
+    private void setBlockFromInventory(@NotNull final BlockPos location, @NotNull final Block block)
+    {
+        worker.swing(worker.getUsedItemHand());
+        setBlockFromInventory(location, block, block.defaultBlockState());
+    }
+
+    private void setBlockFromInventory(@NotNull final BlockPos location, final Block block, final BlockState metadata)
+    {
+        final int slot;
+        if (block instanceof LadderBlock)
+        {
+            slot = worker.getCitizenInventoryHandler().findFirstSlotInInventoryWith(block);
+        }
+        else
+        {
+            slot = worker.getCitizenInventoryHandler().findFirstSlotInInventoryWith(block);
+        }
```

Comment:
ye well this kinda how should I put it seems a liiitle redundant? :D

Reply:
God, the age of this code xD